### DISCUSSION TITLE
ESMIF fine-tuning

### DISCUSSION THREAD

**Discussion by OliviaViessmann** [facebookresearch/esm#450] [Q&A]
Hi,

I would like to fine-tune ESMIF on a subset of alphafolded structures. I forked the repo and set up a training script, however I have some questions:
1. How to best initialise the sequence? I am currently using an initial guess of all "X" (unknown amino acid). With this I see the loss dropping in the beginning of fine-tuning drastically. I expected a fairly flat loss curve given the model is fully trained. Is the it the initialisation or maybe my alphafold dataset subset is out of distribution from original training data?
2. Any suggestions on the span masking? Is there an available function (I couldn't find it).
3. Is there generally training code available that was used for the pretrained ESMIF model?

Many thanks and happy new year 2023,
Olivia
